ext/opcache/jit/zend_jit_trace.c

```diff
@@ -7656,6 +7656,22 @@ static void zend_jit_blacklist_root_trace(const zend_op *opline, size_t offset)
 	zend_shared_alloc_unlock();
 }
 
+ZEND_EXT_API void zend_jit_blacklist_function(zend_op_array *op_array) {
+	zend_jit_op_array_trace_extension *jit_extension = (zend_jit_op_array_trace_extension *)ZEND_FUNC_INFO(op_array);
+	if (!jit_extension || !(jit_extension->func_info.flags & ZEND_JIT_ON_HOT_TRACE)) {
```

Comment:
This is relatively easy for functions that are not JIT-ed yet. See ``zend_jit_op_array()`` in zend_jit.c. It creates different ``zend_jit_op_array*_extension structures for different triggers. The trigger may be determined through corresponding flag in ``jit_extension->func_info.flags``. So for different triggers you should just restore the original handlers. This should work for ZEND_JIT_ON_FIRST_EXEC, ZEND_JIT_ON_PROF_REQUEST (only one handler should be reverted), and ZEND_JIT_ON_HOT_COUNTERS  (all handlers should be reverted).

For ZEND_JIT_ON_SCRIPT_LOAD trigger and op_arrays JIT-ed because of ZEND_JIT_ON_FIRST_EXEC, ZEND_JIT_ON_PROF_REQUEST and ZEND_JIT_ON_HOT_COUNTERS we should use ``zend_jit_op_array_extension`` structure with ZEND_JIT_ON_SCRIPT_LOAD flag and single entry handler. It's possible to create it in ``zend_real_jit_func`` or ``zend_jit``.

Anyway, this looks not trivial and may delay this PR for edges.